### ISSUE TITLE
Bad state: Cannot add new events after calling close

### ISSUE THREAD

**Issue by divarela** [shamblett/mqtt_client#516]
I believe this error occurs when the lib tries to send an ack message after being disconnected.

Stack trace
```
_BroadcastStreamController.add (dart:async)
EventBus.fire (event_bus.dart:61)
PublishingManager.handlePublish (mqtt_client_publishing_manager.dart:165)
MqttConnectionHandlerBase.messageAvailable (mqtt_client_mqtt_connection_handler_base.dart:225)
```

Would be possible to add a validation before send a ack message?

**Comment by shamblett** [comment]
Yes OK, I'll look at this

**Comment by shamblett** [comment]
Updated the client to log when this condition occurs rather than failing as it does now.

Package re published at version 10.2.0

**Comment by saninthottungal** [comment]
Hi, i'm facing the same issue in `mqtt5_client: ^4.1.0`

if i upgrade to the latest version `mqtt5_client: ^4.13.2`, will it fix my issue?

I'm asking because I'm not able to verify if the issue is fixed in the latest version, as it occurs only sometimes and there are no clear steps to reproduce it.
